csrc/ir_nodes.cpp

```diff
@@ -2568,6 +2567,33 @@ TensorDomain::TensorDomain(
   resetDomains();
 }
 
+namespace {
+
+// Validate that the root domain consists of all inputs to domain
+// Uncertain if this will hold for RFactor
+void validateInputDependency(
+    const std::vector<IterDomain*>& root_domain,
+    const std::vector<IterDomain*>& domain) {
+  std::vector<Val*> non_symbolic_domain;
+  std::copy_if(
+      domain.begin(),
+      domain.end(),
+      std::back_inserter(non_symbolic_domain),
+      [](auto dom) { return dom->getIterType() != IterType::Symbolic; });
+  auto inps = IterVisitor::getInputsTo(non_symbolic_domain);
```

Comment:
Nit: I think we should avoid `getInputsTo` when possible. We don't really care if for some reason the ID's have non nullptr definitions. Point being we probably want to make sure the same property is true for rfactor, but rfactor iter domains may still have definitions.

Reply:
Do you mean a root ID may have a definition?

Reply:
I'll update this as a separate PR. Tracked here #225